### DISCUSSION TITLE
OpenWrt 路由器定时测速替换优质 IP 适合所有科学软件

### DISCUSSION THREAD

**Discussion by eightsheep67** [XIU2/CloudflareSpeedTest#311] [分享]
https://github.com/eightsheep67/cf-autoupdate

只需要将CF加速的节点IP替换为路由器本机IP，比如192.168.1.1，端口默认设置为8443即可，还可以推送结果。

**Comment by ghost** [comment]
不错，但如何设置不是定一个时间，而是间隔制？
比如每两小时测一次，同时规定在指定时间段，例如晚20点以后，或晚20至凌晨3时不测试？

还有个问题，路由器设置好后能否关机？重开还有效吗？

**Comment by eightsheep67** [comment]
> 不错，但如何设置不是定一个时间，而是间隔制？ 比如每两小时测一次，同时规定在指定时间段，例如晚20点以后，或晚20至凌晨3时不测试？
> 
> 还有个问题，路由器设置好后能否关机？重开还有效吗？

多设置几个定时不就可以了？而且一直频繁换ip没有意义，CF迁移IP都没有那么频繁，一天设置早晚8点两次差不多了。

重启后需要手动在运行一次脚本建立端口转发，也可以添加到启动项里，我没研究也不会弄……

**Comment by ghost** [comment]
喔，这样啊……
频繁换是因为我这里电信筛选出的隔天就完全挂掉了。。。
甚至离谱的早晨测出来的，到下午4点左右都挂了……我也不想频繁换啊！

**Comment by eightsheep67** [comment]
> 喔，这样啊…… 频繁换是因为我这里电信筛选出的隔天就完全挂掉了。。。 甚至离谱的早晨测出来的，到下午4点左右都挂了……我也不想频繁换啊！

完全挂掉？是连接不上吗，甚至连cloudflare.com网站都打不开吗？
如果是的话，检查一下是否有BT下载的终端也在翻墙，这个之前我有遇到过。这个情况下CF会被切断十几分钟，然后恢复。应该是被CF给Qos了。需要把BT下载的终端单独出来设置不翻墙。

如果只是降速，那摸出大概规律后，就设置那几个时间点就行了。看你描述的也是正常现象啊，我这边一般就是晚上7点到8点需要重新更换IP，但那时候也就是晚高峰换什么都也不会差太多。我公司的V2ray一直用同一个IP，因为是白天用，都没换过速度依然正常。

**Comment by ghost** [comment]
不不。
我openwrt里没装。
是在电脑里筛选，然后用ip来搞v2的。cf网站可以正常直开，速度只是略慢与国内网页。
完全挂掉就是速度会跌的很慢（这种情况在之前有），但是现在基本就是隔天就完全连接不上了，能ping，但就是完全没流量。
家里还是公网ip（动态的，差不多1天一换，习惯晚上睡觉关路由），感觉是从去年底突然一下子不对劲的……之前我还有提交过issues问是不是筛选的问题。
估计如同大神所说，应该是运营商qos了，或者加了啥审查。

用ip一般就youtube，下载，再没干啥。推啊，电报啥的都不玩。。。
我现在v2是在heroku做的，然后用我的网站加个子域名，然后cfworker反代转发过来。v2n里之前服务器地址用的筛选，现在改成用一些国内的cf的dns加速网站他们的地址，有些说是每15分钟自动筛选，我也搞不清原理。。。

反正速度肯定没有自己直接筛选出来快，在出问题之前可以筛出来下载速度25M/S的ip。。然而现在筛选出来的最多10M左右了。。。而国内dns加速套壳的，有时候却可以到15M。不过这些都是白天。。一到晚上下载速度有线我还没测试，无线下国内的可以在9M左右，v2就只有1~2M。大概缩九分之一呢。

总的来说，感觉还是我这地方电信局服务的问题，不过照你说的，会不会cf给我的账号整个限速了？永久的？
因为我cf账号下所有的服务每个月大概能跑200G左右的流量……有时候甚至500G。。。
基本都是翻出去下载东西的时候多……

**Comment by eightsheep67** [comment]
> 不不。 我openwrt里没装。 是在电脑里筛选，然后用ip来搞v2的。cf网站可以正常直开，速度只是略慢与国内网页。 完全挂掉就是速度会跌的很慢（这种情况在之前有），但是现在基本就是隔天就完全连接不上了，能ping，但就是完全没流量。 家里还是公网ip（动态的，差不多1天一换，习惯晚上睡觉关路由），感觉是从去年底突然一下子不对劲的……之前我还有提交过issues问是不是筛选的问题。 估计如同大神所说，应该是运营商qos了，或者加了啥审查。
> 
> 用ip一般就youtube，下载，再没干啥。推啊，电报啥的都不玩。。。 我现在v2是在heroku做的，然后用我的网站加个子域名，然后cfworker反代转发过来。v2n里之前服务器地址用的筛选，现在改成用一些国内的cf的dns加速网站他们的地址，有些说是每15分钟自动筛选，我也搞不清原理。。。
> 
> 反正速度肯定没有自己直接筛选出来快，在出问题之前可以筛出来下载速度25M/S的ip。。然而现在筛选出来的最多10M左右了。。。而国内dns加速套壳的，有时候却可以到15M。不过这些都是白天。。一到晚上下载速度有线我还没测试，无线下国内的可以在9M左右，v2就只有1~2M。大概缩九分之一呢。
> 
> 总的来说，感觉还是我这地方电信局服务的问题，不过照你说的，会不会cf给我的账号整个限速了？永久的？ 因为我cf账号下所有的服务每个月大概能跑200G左右的流量……有时候甚至500G。。。 基本都是翻出去下载东西的时候多……

你也说你在下载了，那肯定会被CF限速的。

**Comment by XIU2** [comment]
@Visntse 如果是 Cloudflare 限速/限制你的域名/账号，那么肯定会发邮件告诉你的（像我这种情况直到一个月几百 TB 的情况下才被 Cloudflare 发现并限速了），不会偷偷摸摸的干。

**Comment by eightsheep67** [comment]
> @Visntse 如果是 Cloudflare 限速/限制你的域名/账号，那么肯定会发邮件告诉你的（像我这种情况直到一个月几百 TB 的情况下才被 Cloudflare 发现并限速了），不会偷偷摸摸的干。

会被临时QoS吧，我这边电信实测，如果我在翻墙电脑上开迅雷下BT，一定会cloudflare掉线，停掉迅雷几分钟后恢复，再开在掉线，不开就没有掉线过。我的情况就是直接Cloudflare.com都打不开，所有CF的IP地址包括由CF加速的网站统统打不开。这个感觉不是针对你的账号，是针对你的IP。

**Comment by ghost** [comment]
之前用大神给的一个工具测试过，反正我这边家里到电信局会丢包，再到北京出口还会丢包，简直了……
也不知道是啥问题。
cf邮件倒是没收过，也不知道是啥问题，现在用v2,xray,tr-go都不行。。tr-go大前天布局了一下，速度奇慢，v2没有xr快，于是前天重布了xr，然后用vless，但是这两天晚上下载就只有400多kb了。。比之前更慢了。
唉算了。。就这么着吧，反正都是白嫖的。
对了额外问一下，ow筛选和广告屏蔽软件有没有冲突？
我的ow里装着路由/NAT分载，在防火墙里设置着，软硬件流量双提速的。还用着simple-adblock看设置似乎使用dnsmaq-full来屏蔽广告的，都是官方的ipk。如果把v2和你这个装上，互相冲突吗？

**Comment by eightsheep67** [comment]
> 之前用大神给的一个工具测试过，反正我这边家里到电信局会丢包，再到北京出口还会丢包，简直了…… 也不知道是啥问题。 cf邮件倒是没收过，也不知道是啥问题，现在用v2,xray,tr-go都不行。。tr-go大前天布局了一下，速度奇慢，v2没有xr快，于是前天重布了xr，然后用vless，但是这两天晚上下载就只有400多kb了。。比之前更慢了。 唉算了。。就这么着吧，反正都是白嫖的。 对了额外问一下，ow筛选和广告屏蔽软件有没有冲突？ 我的ow里装着路由/NAT分载，在防火墙里设置着，软硬件流量双提速的。还用着simple-adblock看设置似乎使用dnsmaq-full来屏蔽广告的，都是官方的ipk。如果把v2和你这个装上，互相冲突吗？

我自己编译固件的话，都是插件越少越好。广告屏蔽那类插件没用了，容易出问题。

**Comment by sxml** [comment]
大神 那个passwall专用 能修改SSR用吗？扫IPV6 通用版不能用

**Comment by s1u** [comment]
没太看懂是干嘛用的，能稍微解释下是什么功能？
